src/style/NewBranchDialog.ts

```diff
@@ -0,0 +1,246 @@
+    }
+  }
+});
+
+export const nameInputClass = style({
+  boxSizing: 'border-box',
+
+  width: '100%',
+  height: '2em',
+
+  marginBottom: '16px',
+
+  /* top | right | bottom | left */
+  padding: '1px 18px 2px 7px',
+
+  fontSize: 'var(--jp-ui-font-size1)',
+  fontWeight: 300,
+
+  border: 'var(--jp-border-width) solid #e0e0e0',
+  borderRadius: '3px',
+
+  $nest: {
+    '&:active': {
+      border: 'var(--jp-border-width) solid var(--jp-brand-color1)'
+    },
+    '&:focus': {
+      border: 'var(--jp-border-width) solid var(--jp-brand-color1)'
+    }
+  }
+});
+
+export const filterWrapperClass = style({
+  padding: 0,
+  paddingBottom: '4px'
+});
+
+export const filterClass = style({
+  boxSizing: 'border-box',
+  display: 'inline-block',
+  position: 'relative',
+
+  width: '100%',
+
+  marginRight: '11px',
+
+  fontSize: 'var(--jp-ui-font-size1)'
+});
+
+export const filterInputClass = style({
+  boxSizing: 'border-box',
+
+  width: '100%',
+  height: '2em',
+
+  /* top | right | bottom | left */
+  padding: '1px 18px 2px 7px',
+
+  fontSize: 'var(--jp-ui-font-size1)',
+  fontWeight: 300,
+
+  border: 'var(--jp-border-width) solid #e0e0e0',
+  borderRadius: '3px',
+
+  $nest: {
+    '&:active': {
+      border: 'var(--jp-border-width) solid var(--jp-brand-color1)'
+    },
+    '&:focus': {
+      border: 'var(--jp-border-width) solid var(--jp-brand-color1)'
+    }
+  }
+});
+
+export const filterClearClass = style({
+  position: 'absolute',
+  right: '5px',
+  top: '0.6em',
+
+  height: '1.1em',
+  width: '1.1em',
+
+  padding: 0,
+
+  backgroundColor: '#757575',
+
+  border: 'none',
+  borderRadius: '50%',
+
+  $nest: {
+    svg: {
+      width: '0.5em!important',
+      height: '0.5em!important',
+
+      fill: 'white'
+    },
+    '&:hover': {
+      backgroundColor: '#616161'
+    },
+    '&:active': {
+      backgroundColor: '#424242'
+    }
+  }
+});
+
+export const listWrapperClass = style({
+  boxSizing: 'border-box',
+  display: 'block',
+
+  width: '100%',
+  height: '200px',
+
+  border: 'var(--jp-border-width) solid #e0e0e0',
+  borderRadius: '3px',
+
+  overflow: 'hidden',
+  overflowY: 'scroll'
+});
+
+export const listItemClass = style({
+  flexDirection: 'row',
+  flexWrap: 'wrap',
+
+  width: '100%',
+
+  /* top | right | bottom | left */
+  padding: '4px 11px 4px 11px!important',
+
+  fontSize: 'var(--jp-ui-font-size1)',
+  lineHeight: '1.5em'
+});
+
+export const activeListItemClass = style({
+  color: 'white!important',
+
+  backgroundColor: 'var(--jp-brand-color1)!important'
+});
+
+export const listItemContentClass = style({
+  flexBasis: 0,
+  flexGrow: 1,
+
+  marginTop: 'auto',
+  marginBottom: 'auto',
+  marginRight: 'auto'
+});
+
+export const listItemDescClass = style({
+  marginBottom: 'auto'
+});
+
+export const listItemIconClass = style({
+  width: '16px',
+  height: '16px',
+
+  /* top | right | bottom | left */
+  margin: 'auto 8px auto 0',
+
+  backgroundImage: 'var(--jp-icon-git-branch-light-theme)',
```

Comment:
We should prefer the svg injection method for new icon.

See #515 for an example on how to add and use a new one.

Reply:
I would argue against this. SVG injection just results in more DOM nodes and worse application performance. Better to use CSS image assets and allow the browser to leverage caching.

Reply:
> SVG injection just results in more DOM nodes and worse application performance

@kgryte Well, that opinion is worrisome. Since 99% of the icons in core are now rendered as svg nodes, and since I am 100% responsible for that change, I could use a bit more guidance as to your strong stance on the issue. When I first switched the icons over to the icons-as-svg-nodes pattern I did do a bunch of profiling. I did not see any change in performance that was significant relative to the noise. 

Since I know that you know more than me about the deep mysteries of JS, I am a bit concerned. How grounded in your own experience is your opinion on the matter? Is the icons-as-svg-nodes pattern an inherently Bad Idea, or are you just operating on the general principle that more nodes => more overhead => more bad?

From my viewpoint, the big advantage that the icons-as-svg-nodes pattern has over the older icons-as-css-background pattern is that svg nodes are reachable by css. Given the right setup, this allows icons to be dynamically repainted when the theme changes with the actual colors provided by the particular theme. As far as I've been able to dig up, there doesn't seem to be another good way of dynamically and flexibly theming icons

Reply:
@telamonian You are right. When it comes to theming as is wanted in JupyterLab, you don't have much of a choice apart from loading different theme assets via CSS (as is currently down now in this PR for dark and light modes).

However, one needs to be careful in ensuring that the use of SVG icons does not cause the DOM tree to explode. For example, were we to use SVG injection here (for branch icons) and we had `1000` branches, assuming, for a moment, no virtual scrolling/windowing, then we'd inject 1000 SVG icons into the DOM. Upon doing so, performance, in particular DOM tree traversal, will be adversely affected based on the principle that, yes, more node => more overhead => more bad.

That this principle affects JupyterLab more generally is supported by the fact that JupyterLab exhibits poor performance for large notebooks (e.g., due to the number of DOM nodes associated with each input and output cell).

In this case, particularly when an image is going to be repeated many times within a document, leveraging CSS is better (perf-wise), despite being more limited/restrictive for theming.

For what you are doing with JLab where you can expect, e.g., toolbar icons to render once, or at most a handful times, in the application, using SVG icons is fine. Issues arise, however, when SVG nodes proliferate within an HTML document--a point to which the web data visualization community can readily attest.

Reply:
Hmm, interesting, and something I definitely need to keep an eye out for. Thanks for the info. From what you say, I think the only likely pain point in core is then going to be the filebrowser. I know it's fine up to at least ~2000 items, but I can see how it might spiral out past that. Some basic pagination is probably in order.

There's also some more general possible optimizations that I've been putting off until I had reason to no longer think they'd be premature. There's an interesting one using `xmlns:xlink` that can supposedly save cycles when rendering an icon in multiple places.

Reply:
@telamonian Yeah, I think that windowing/virtual scrolling could be an effective solution in the filebrowser to minimize the number of DOM nodes.